packages/nuxt/src/app/components/nuxt-link.ts

```diff
@@ -59,6 +59,11 @@ export interface NuxtLinkProps extends Omit<RouterLinkProps, 'to'> {
    * When enabled will prefetch middleware, layouts and payloads of links in the viewport.
    */
   prefetch?: boolean
+  /**
+   * Determines when to prefetch links. Defaults to `on-visible`. This prop is only used when `prefetch` is
+   * enabled (default) and `noPrefetch` is not set.
+   */
+  prefetchMode?: 'on-hover' | 'on-visible'
```

Comment:
This is just a personal opinion, but would it be better to use `visible` and `hover`?

Reply:
Thanks for the suggestion, I made it this way because it's more explicit from my viewpoint. `prefetchMode`: 'on-hover".  It sounds for me like "When? On hover".

Reply:
I would like to share my thoughts on this matter. When we use native event listeners, we do it like this:

```ts
element.addEventListener('pointerenter', callbackHandler)
```

It does not start with `on`. I simply believe that using `visible` and `hover` is more in line with what I assume to be common practice.

However, this is just my personal opinion. Regardless of the API you ultimately choose, I am grateful for your and the core team's hard work.

Reply:
how about:

```suggestion
  prefetchMode?: 'hover' | 'visibility'
```

Reply:
@danielroe makes sense